tests/testing.py

```diff
@@ -0,0 +1,312 @@
+        cls.delegator_wallet = LocalWallet(PrivateKey(delegator_bip44_def_ctx.PrivateKey().Raw().ToBytes()))
+        cls.delegator_address = str(cls.delegator_wallet.address())
+
+        cfg = NetworkConfig(
+            chain_id="testing",
+            url="grpc+http://localhost:9090",
+            fee_minimum_gas_price=1,
+            fee_denomination="atestfet",
+            staking_denomination="atestfet",
+        )
+
+        gov_cfg = grpc.insecure_channel('localhost:9090')
+
+        cls.ledger_client = LedgerClient(cfg)
+        cls.gov_module = query_pb2_grpc.QueryStub(gov_cfg)
+
+        transport = AIOHTTPTransport(url="http://localhost:3000")
+        cls.gql_client = Client(transport=transport, fetch_schema_from_transport=True)
+
+        cls.db = psycopg.connect(
+            host="localhost",
+            port="5432",
+            dbname="postgres",
+            user="postgres",
+            password="postgres",
+            options=f'-c search_path=app'
+        )
+
+        cls.db_cursor = cls.db.cursor()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.db.close()
+
+
+class GovernanceTests(Base):
+    vote_tx = None
+    denom = "atestfet"
+    amount = "10000000"
+    option = "YES"
+
+    def setUp(self):
+        self.msg = gov_tx.MsgVote(
+            proposal_id=1,
+            voter=self.validator_address,
+            option=gov_pb2.VoteOption.VOTE_OPTION_YES
+        )
+        self.vote_tx = Transaction()
+        self.vote_tx.add_message(self.msg)
+
+    def testGovernanceProposal(self):
+        proposal_content = any_pb2.Any()
+        proposal_content.Pack(gov_pb2.TextProposal(
+            title="Test Proposal",
+            description="This is a test proposal"
+        ), "")
+
+        msg = gov_tx.MsgSubmitProposal(
+            content=proposal_content,
+            initial_deposit=[coin_pb2.Coin(
+                denom=self.denom,
+                amount=self.amount
+            )],
+            proposer=self.validator_address
+        )
+
+        tx = Transaction()
+        tx.add_message(msg)
+
+        tx = utils.prepare_and_broadcast_basic_transaction(self.ledger_client, tx, self.validator_wallet)
+        tx.wait_to_complete()
+        self.assertTrue(tx.response.is_successful())
+        # response = self.gov_module.Proposal(gov_query.QueryProposalRequest(proposal_id=1))
+
+    def testProposalVote(self):
+        self.db_cursor.execute('TRUNCATE table gov_proposal_votes')
+        self.db.commit()
+        self.assertFalse(self.db_cursor.execute('SELECT * from gov_proposal_votes').fetchall())
+
+        tx = utils.prepare_and_broadcast_basic_transaction(self.ledger_client, self.vote_tx, self.validator_wallet)
+        tx.wait_to_complete()
+        self.assertTrue(tx.response.is_successful())
+
+        time.sleep(5)
+
+        row = self.db_cursor.execute('SELECT * from gov_proposal_votes').fetchone()
```

Comment:
nit: Can we pull the SQL query out into a variable, I think this would improve maintainability.